components/public-api/gitpod/experimental/v1/workspaces.proto

```diff
@@ -147,6 +150,9 @@ message WorkspaceContext {
     message Git {
         string normalized_context_url = 1;
         string commit = 2;
+
+        // gitProvider is the git provider, e.g. 'github', 'gitlab', self hosted, etc
+        string gitProvider = 3;
```

Comment:
Only if it's really easy to add.
This is what I'm currently doing client side to infer ir
```
    const toWorkspaceData = (ws: Workspace) => {
        const url = new URL(ws.context!.contextUrl);
        const provider = url.host.replace(/\..+?$/, ''); // remove '.com', etc
        const matches = url.pathname.match(/[^/]+/g)!; // match /owner/repo
        const owner = matches[0];
        const repo = matches[1];
        return {
            provider,
            owner,
            repo,
            id: ws.workspaceId,
            contextUrl: ws.context!.contextUrl,
            workspaceUrl: ws.status!.instance!.status!.url,
            phase: ws.status?.instance?.status?.phase
        };
    };
```